src/short-urls/reducers/shortUrlCreation.ts

```diff
@@ -8,31 +8,18 @@ import { createAsyncThunk, useApiClientFactory } from '../../store/helpers';
 
 const REDUCER_PREFIX = 'shlink/shortUrlCreation';
 
-// TODO Migrate this to `status: 'idle' | 'saving' | 'saved' | 'error'`
 export type ShortUrlCreation = {
-  saving: false;
-  saved: false;
-  error: false;
-} | {
-  saving: true;
-  saved: false;
-  error: false;
-} | {
-  saving: false;
-  saved: false;
-  error: true;
-  errorData?: ProblemDetailsError;
+  status: 'idle' | 'saving';
 } | {
+  status: 'saved';
   result: ShlinkShortUrl;
-  saving: false;
-  saved: true;
-  error: false;
+} | {
+  status: 'error';
+  error?: ProblemDetailsError;
```

Comment:
Renaming `errorData` to `error`, as there's now no conflict with the previous `error` boolean flag.